app/domain/authentication/authn_jwt/validate_and_decode/get_verification_option_by_jwt_claim.rb

```diff
@@ -0,0 +1,61 @@
+module Authentication
+  module AuthnJwt
+    module ValidateAndDecode
+      # GetVerificationOptionByJwtClaim command class is responsible to get jwt claim and return his verification option,
+      # in order to validate it against JWT 3rd party, for example:
+      # 1. Input: {name: iss, value: cyberark.com} // jwt claim
+      #    Output: {:iss => cyberark.com, :verify_iss => true} // verification option dictionary
+      # 2. Input: {name: iat, value: } // jwt claim
+      #    Output: {:verify_iat => true} // verification option dictionary
+      # 3. Input: {name: exp, value: } // jwt claim
+      #    Output: {} // verification option dictionary
+      # 4. Input: {name: nbf, value: } // jwt claim
+      #    Output: {} // verification option dictionary
+      GetVerificationOptionByJwtClaim ||= CommandClass.new(
+        dependencies: {
+          logger: Rails.logger
+        },
+        inputs: [:jwt_claim]
+      ) do
+        def call
+          validate_claim_exists
+          get_verification_option_by_jwt_claim
+        end
+
+        private
+
+        def validate_claim_exists
+          raise Errors::Authentication::AuthnJwt::MissingClaim if @jwt_claim.blank?
+        end
+
+        def get_verification_option_by_jwt_claim
+          claim_value = @jwt_claim.value
+          claim_name = @jwt_claim.name
```

Comment:
these can be transferred to memoizing methods

Reply:
Done